### PR TITLE
🎨 : Introduced Adaptive alert action for adaptive dialog.

### PR DESCRIPTION
**Pull request by aliasgar4558**
**Introducing new widget for displaying adaptive action buttons for adaptive alert dialog**

Issue : https://github.com/flutter/flutter/issues/134347

- NO breaking changes introduced in the changes, just new widget added for the community.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/lib/src/material/dialog.dart [hunk 1]
```diff
@@ -951,6 +952,68 @@ class _AdaptiveAlertDialog extends AlertDialog {
...
+/// The [AdaptiveAlertAction] class is a stateless widget used to create action buttons
+/// for alert dialogs or similar user interface components. It takes a [child] parameter
+/// to specify the button's content and an [onPressed] callback to define the action when
+/// the button is pressed.
+///
+/// Example usage:
+///
+/// ```dart
+/// AdaptiveAlertAction(
+///   child: Text('OK'),
```
Comment (MitchellGoodwin) [inline packages/flutter/lib/src/material/dialog.dart:966]: The linter tests are complaining about this line. They still check example code, and the `Text` widget needs to be a const.
```suggestion
///   child: const Text('OK'),
```

Comment (aliasgar4558) [reply packages/flutter/lib/src/material/dialog.dart]: Added `const` against the Text widget in example code.

**Comment by MitchellGoodwin** [comment]
@aliasgar4558 could you have the account for @rspl-aliasgar sign the CLA as well? Context tells me it's you, but it can be tricky legally if all accounts tied to commits to a PR haven't signed it.

**Comment by aliasgar4558** [comment]
@MitchellGoodwin - you are right, it's me , but I mistakenly commited the change with different email (user) but I have already signed the CIA on that account as well. Don't know why it didn't reflect. You got any idea for that? Or just let me know if someone I can revert the commit and commit again with my primary account (email)....

**Comment by MitchellGoodwin** [comment]
I triggered the rescan and it caught it. 👍

**Comment by aliasgar4558** [comment]
> I triggered the rescan and it caught it. 👍

Great - Thank you so much mate.

**Comment by MitchellGoodwin** [comment]
Thank you for taking the time to put together this contribution, but after speaking with others on the team, we do not think this is the direction we want to go, as we do not want to enforce any sort of implied default, and if the developer wants their behavior it's simple to write themselves. If you are interested in taking this further, please feel free to make a package providing it.

**Pull request closed without merging**